05 Iteration

zentox opened the pull request:
Small enhancements and new features:
- Added an option to add/remove courses as favourite
- Display favourite on the profile of the user
- Hyperlinks to the table of content for the course
- Embedding graphics in LaTeX and an information how to use it as tooltip
- Translations
- Reverting courses and content of a course
- Creating topics in the course structure
- Adding existing topics to the course structure
- Drag & Drop option for topics to reorder topics in the course structure

LastNonak committed:
moved Latex Compile Time

Latex Compiling now occurs *after* the attachments are evaluation of the attachments, because the attachments need to be available at compile time

LastNonak committed:
add latex embeddings

images can now be embedded into a latex document

LastNonak committed:
added TODO

added TODO in expanded method

LastNonak committed:
adapted Regex to allow scaling

Regex now allows options for Images

LastNonak committed:
changed r +f to rf

combined raw and regex expression

LastNonak committed:
added "save" function

save function works now:
- users can save and unsave courses
- favourite courses now available in profile detail view

LastNonak committed:
added attachment helper

user can now click on a question mark next to the latex field in order to be informed about how to embed images in their latex document

LastNonak committed:
changed '?' to fontawesome icon

usage of fontawesome question mark instead of '?'

LastNonak committed:
adapted css of popup

popup opens now directly on top of the question mark instead of to the right

LastNonak committed:
consistency of stared courses <=> favourite courses

- stared courses layout in profile now consistent with stared courses layout on dashboard
- renamed favourite courses in profile to stared courses

LastNonak committed:
consistency: " vs ' in HTML

" and ' now consistent in HTML data

zentox committed:
Merge branch '05_attachment_embedding' into 05_base

zentox committed:
Updated compatibility with review 04 code

zentox committed:
Merge branch '05_course_structure' into 05_base

zentox committed:
Merge branch '05_vc' into 05_base

Dieser-Typ committed:
set type in revert

zentox committed:
Update history.py

lizz-py committed:
Update README.md

zentox committed:
Fixed css header base conflicts with structure

zentox committed:
Merge branch '05_base' of https://github.com/BP-Collab-Coursebook/collab-coursebook into 05_base

lizz-py committed:
impoved scaling image for export

lizz-py committed:
Merge branch '05_base' of https://github.com/BP-Collab-Coursebook/collab-coursebook into 05_base

lizz-py committed:
Update error.tex

lizz-py committed:
added example to explanation of embedded images in latex

LastNonak committed:
add vertical dividing line

vertical dividing line added in edit_structure view

lizz-py committed:
translation for courses done, removed the line in edit structure for now because of the bug

lizz-py committed:
translation for edit course structure

zentox committed:
Separate forms in edit-structure - CSS

zentox committed:
Hotfix frontend view bug with title image in content - edit form

zentox committed:
Update setup.sh

Added reversion to setup

lizz-py committed:
finished translations

lizz-py committed:
Merge branch '05_base' of https://github.com/BP-Collab-Coursebook/collab-coursebook into 05_base

lizz-py committed:
changed to version so there is no need for translation

zentox committed:
PyLint + Warnings - base

zentox committed:
PyLint + Warnings - content

zentox committed:
PyLint + Warnings - export

zentox committed:
Update admin.py

zentox committed:
PyLint + Warnings - frontend

zentox committed:
PyLint + Warnings - test

zentox committed:
Update PyLint disable

zentox committed:
Refactored history function based view to class based view - post request

zentox committed:
Small adjustment of removing redundant code and fixing docs

zentox committed:
Update template names

zentox committed:
Update docstring - base

zentox committed:
Refactored code, adjustment docstring, PyLint

zentox committed:
HTML code indention - content

zentox committed:
HTML code indention - frontend/comment/

zentox committed:
HTML code indention - frontend/content/

zentox committed:
templates added empty line - content

zentox committed:
HTML code indention - frontend/course/

zentox committed:
HTML code indention - frontend/course_lists/

zentox committed:
HTML code indention - frontend/coursebook/

zentox committed:
HTML code indention - frontend/history/

zentox committed:
HTML code indention - frontend/profile/

zentox committed:
HTML code indention - frontend/widgets/

zentox committed:
HTML code indention - frontend/

Lian-zzz committed:
- Fix translation for revert, 'Edit Structure', course period sort, 'coursebook' etc.

zentox committed:
Update pylint.yml

Increased minimum threshold to 9

zentox committed:
Update content.py

Dieser-Typ committed:
Update compare

fixed bug with annoying question marks in diff

Dieser-Typ committed:
Compare for Courses

Fix Bug with question marks also for courses

zentox committed:
Update history.py

Added comment to understand the code

zentox committed:
Merge branch '05_base' into review_05

lizz-py committed:
fixed translations errors

lizz-py committed:
change log order changed, need to fix issue for edit content change log

lizz-py committed:
changed order for change log

zentox committed:
Removed nestable references to vendor and adjust path of needed files for nestable

lizz-py committed:
fixed translation of changelog in contents

lizz-py committed:
Merge branch 'review_05' of https://github.com/BP-Collab-Coursebook/collab-coursebook into review_05

lizz-py committed:
changed the symbol for revert

LastNonak committed:
favourite courses only viewable in own profile

a user cannot see the marked courses of other users anymore

zentox committed:
Hotfix nestable empty json object bug

Adjustment on parse function to break if the json object is empty

lizz-py committed:
hotfix post edit course structure

There was an error, when you tried to filter topics in a newly set topic in edit course structure

zentox committed:
Hotfix missing post request for sorting/filtering

LastNonak committed:
fixed name of export file

export file now named "coursebook"

LastNonak committed:
title of pdf export file translated

title of pdf export file now also translated

zentox committed:
Hotfix empty nestable list bug with drag & drop

zentox committed:
Removed add topic with same name validator since there can exists topic with the same name but different category

Lian-zzz committed:
- test for textfield compare & revert

Lian-zzz committed:
- added tests for json
- one little bug fixed for validate_topics
- minor change for CourseStructureEntry's document
- test for content history restructured
- added tests for latex history with different cases

Lian-zzz committed:
- added tests for views/course
- added tests for course history

Lian-zzz committed:
- hotfix

Lian-zzz committed:
- more cases added
- bugfix for CourseView

zentox committed:
Review json test

zentox committed:
Update test_json.py

zentox committed:
Review history test

zentox committed:
Update test_json.py

zentox committed:
Update edit_structure.html

Lian-zzz committed:
update test_course and test_json:
- for the loops tests: when more than 1 element is relevant, there are now more elements rather than 2 to be executed
- no methods from same test class
- cases seperated for test_course

Lian-zzz committed:
update test_course and test_json:
- for the loops tests: when more than 1 element is relevant, there are now more elements rather than 2 to be executed
- no methods from same test class
- cases seperated for test_course

Lian-zzz committed:
add todo

Lian-zzz committed:
- add todo
- docu modified

Dieser-Typ committed:
Update test_course.py, json.py

Fix typos and doc and remove print statements from json.py
move BaseCourseViewTestCase to test_cases.py and resolve redundant code

Dieser-Typ committed:
resolve redundancies and pylint warnings

Dieser-Typ committed:
pylint

zentox committed:
Review pylint

zentox committed:
Merge branch 'review_05_test' into review_05

zentox committed:
Review PyLint

lizz-py commented on requirements.txt.py:
```diff
@@ -0,0 +1,12 @@
+XXXXXXXXXXXXX
```
What are these symbols?

zentox replied:
This file should not have exited and will be removed.

lizz-py reviewed (approved):
Maybe delete all the line numbers in translation, since we are updating code in every iteration and the lines are being shifted. Or just update them at the end of the project.

content/templates/content/view/invalid.html: fehlt eine Übersetzung

export/helper_functions.py : pre_render has a typo in type no_error and there is 2-times type no_error instead of one time type and one time parameter

Help-Text of Latex Image Attachments could be more detailled

HTML change " to ' in all files (mostly for icons or trans)

Remove the TODO comments in some files

zentox commented:
> Maybe delete all the line numbers in translation, since we are updating code in every iteration and the lines are being shifted. Or just update them at the end of the project.

Will be discussed in the next iteration how that will be resolved.

> content/templates/content/view/invalid.html: fehlt eine Übersetzung

Fixed.  62ce7b2

> export/helper_functions.py : pre_render has a typo in type no_error and there is 2-times type no_error instead of one time type and one time parameter

Fixed.  62ce7b2

> Help-Text of Latex Image Attachments could be more detailled

Will be resolved in the next iteration.

> HTML change " to ' in all files (mostly for icons or trans)

Fixed.  62ce7b2

> Remove the TODO comments in some files

Fixed.  62ce7b2
